### PR TITLE
Dyt.cleanup

### PR DESCRIPTION
**Pull request by duanyutong**
### Background

cleanup

### What's new
- ✅ added pre-commit hooks with improved configs
- ✅ cleaned up `CMakeLists`
- ✅ removed cmake modules that can be found in system installation
- ✅ removed filesystem external repo, gcc7 provides fs support as experimental, and gcc8 fully supports it
- ✅ re-wrote `include/util_io.h`, now passed all analysers, completely warning-free
- ✅ added minimal dev setup for pre-commit hooks
- ✅ updated `README`
- ❌ New feature is accompanied by new test: [name and description of new test].

### Related work
- ❌ This PR needs [link]
- ❌ [link] needs this PR

### TODOs / Nice-To-Haves
- ❌ [Add new unit test for new feature.]
- ❌ [Add new system test for new feature.]

### Tests
1. ✅ Tested on simulated robot: `back_and_forth.sh` 10 cycles
2. ❌ Tested on physical robot: [Describe the tests/commands used.]

### Quality
3. ✅ New code is written in pure C++17 to the best of my knowledge.
4. ✅ New code follows established C++17 best practices ([C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines)).
5. ✅ New code passed `clang-format`, `cpplint` at least, **and all** of our static code analysers.
6. ✅ I have included Doxygen-style documentation in the new C++ code.

### Note to reviewers
Please verify that all sections are accurately filled out and that all 6 numbered entries above are present and ticked/checked off, with their requirements met, if applicable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,31 +2,51 @@
...
+```
+
+## Builidng the driver
+
+```bash
+git clone --recursive https://github.com/DexaiRobotics/drake-franka-driver.git`
+drake-franka-driver/setup.sh
+```
+
+See the beginning of `setuo.sh` for more flags.
```
Comment (gavincangan) [inline README.md:32]: `setup.sh`

Comment (dmsj) [inline README.md:32]: setup.sh

**Review by dmsj — COMMENTED** [review]
looks good - just need to test in sim

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,108 +3,76 @@ project(drake-franka-driver)
...
+    src/send_stop.cc
 )
 
+target_link_libraries(send_stop
+    lcm
+)
 
-set(SOURCES 
+# build the lcm driver that uses Franka Robot Interface (FRI)
+set(driver_name franka_driver)
```
Comment (5yler) [inline CMakeLists.txt:74]: Is this used for anything?

**Review by 5yler — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (duanyutong) [reply README.md]: fixed

Comment (duanyutong) [reply README.md]: fixed

File: CMakeLists.txt (hunk 2 above)
Comment (duanyutong) [reply CMakeLists.txt:74]: yes `franka_driver` is the main executable that runs when `franka.sh` is called

**Comment by duanyutong** [comment]
merging after tested with sim robot, 10 cycles with `back_and_forth.sh`

**Pull request merged**